pkg/client/linstor.go

```diff
@@ -1317,15 +1317,27 @@ func (s *Linstor) reconcileSnapshotResources(ctx context.Context, snapshot *volu
 
 	// Optimize the node we use to restore. It should be one of the preferred nodes, or just the first with a snapshot
 	// if no preferred nodes match.
-	selectedNode := snap.Nodes[0]
-
+	var selectedNode string
 	for _, snapNode := range snap.Nodes {
+		if err := s.NodeAvailable(ctx, snapNode); err != nil {
+			logger.WithField("selected node candidate", snapNode).WithError(err).Debug("node is not available")
+			continue
+		}
+
+		if selectedNode == "" {
+			selectedNode = snapNode
+		}
+
 		if slice.ContainsString(preferredNodes, snapNode) {
 			selectedNode = snapNode
 			break
 		}
```

Comment:
could you make the first `if` part of an `else` in the second `if`, I think this makes it easier to read. as in

```go
if slice.ContainsString(...) {
    // we found a perfect candidate
    ...
    break
} else if selectedNode == "" {
   // better than nothing
   ....
}
```

Reply:
I agree. Updated based on your suggestion